runtime/compiler/optimizer/StringPeepholes.cpp

```diff
@@ -2328,3 +2324,21 @@ TR::TreeTop *TR_StringPeepholes::searchForToStringCall(TR::TreeTop *tt, TR::Tree
    return tt;
    }
 
+void TR_StringPeepholes::postProcessTreesForOSR(TR::TreeTop *startTree, TR::TreeTop *endTree)
+   {
+   if (comp()->supportsInduceOSR() &&
+       comp()->isOSRTransitionTarget(TR::postExecutionOSR) &&
+       comp()->getOSRMode() == TR::voluntaryOSR)
+      {
+      if (trace())
+         {
+         TR::Node* startNode = startTree->getNode();
+         TR::Node* endNode = endTree->getNode();
+         traceMsg(comp(), "Post process Trees from %p n%dn to %p n%dn for OSR\n", startNode, startNode->getGlobalIndex(), endNode, endNode->getGlobalIndex());
+         }
+
+      TR::TransformUtil::removePotentialOSRPointHelperCalls(comp(), startTree, endTree);
+      TR::TransformUtil::prohibitOSROverRange(comp(), startTree, endTree);
+      TR::TransformUtil::prohibitGuardedStaticFinalFieldFoldingOverRange(comp(), startTree, endTree);
```

Comment:
I don't think this is necessary - we can fold finals if we can protect the OSR points